opentelemetry-sdk/src/error.rs

```diff
@@ -0,0 +1,45 @@
+use std::sync::PoisonError;
+
+#[cfg(feature = "logs")]
+use crate::logs::LogError;
+#[cfg(feature = "metrics")]
+use crate::metrics::MetricError;
+use opentelemetry::propagation::PropagationError;
+#[cfg(feature = "trace")]
+use opentelemetry::trace::TraceError;
+
+/// Wrapper for error from both tracing and metrics part of open telemetry.
```

Comment:
Though this is moving from just one crate to another, I'd like to first confirm that these are indeed required. I understand that it is desirable to return specialized errors from user-facing APIs, but most of the errors here are only for internal logging only. Is there any additional value provided by this?

Reply:
See https://github.com/open-telemetry/opentelemetry-rust/pull/2241#discussion_r1815986714 for prior discussion. 
If we don't plan to keep this, then its best to just remove it, instead of moving from api to sdk and then removing it again.
(Note: These are still public APIs in the sdk crate)